### ISSUE TITLE
[Firefox] Incoming cookies from requests allowed by exception rules (@@) are not blocked

### ISSUE THREAD

**Issue by cqx931** [dhowe/AdNauseam#1011]
@cqx931  Wasn't this recently checked here: https://github.com/dhowe/AdNauseam/issues/986 ?

**Comment by cqx931** [comment]
only checked everything in chrome for #986

**Comment by cqx931** [comment]
I just verified the issue. 
The problem is not about adnauseam [Allowed!] requests, but how cookies are handled for requests that are allowed according to exception rules.
http://lab-lamp.scm.cityu.edu.hk/adntest/incomingCookies.html
Because we recently added this rule `@@||serving-sys.com^$third-party` in adnaueam.txt, the old incomingCookies test case is now allowed by adnauseam.txt -> case B.
This will directly return 'false' and won't run to case D, where allowRequests got registered.
Should we also register the requests for case B? So that the cookies can also be handled.

Now for case B, the cookies are not accepted in chrome/opera and only appears in FF.
I'm not sure why they got 'blocked' in chrome browsers...but they are definitely not handled by the cookie blocking function we have in adnauseam.
